easybuild/framework/easyconfig/tweak.py

```diff
@@ -795,7 +802,107 @@ def map_toolchain_hierarchies(source_toolchain, target_toolchain, modtool):
     return tc_mapping
 
 
-def map_easyconfig_to_target_tc_hierarchy(ec_spec, toolchain_mapping, targetdir=None):
+def map_versionsuffixes_cache(func):
+    """Function decorator to cache (and retrieve cached) versionsuffixes mapping between toolchains."""
+    cache = {}
+
+    @functools.wraps(func)
+    def cache_aware_func(software_name, software_version_stub, original_toolchain, toolchain_mapping):
+        """Look up original_toolchain in cache first, determine and cache it if not available yet."""
+        # No need for toolchain_mapping to change to be part of the key, it is unique in this context
+        cache_key = (software_name, software_version_stub, original_toolchain['name'], original_toolchain['version'])
+
+        # fetch from cache if available, cache it if it's not
+        if cache_key in cache:
+            _log.debug("Using cache to return version suffix mapping for toolchain %s: %s", str(cache_key),
+                       cache[cache_key])
+            return cache[cache_key]
+        else:
+            versionsuffix_mappings = func(software_name, software_version_stub, original_toolchain, toolchain_mapping)
+            cache[cache_key] = versionsuffix_mappings
+            return cache[cache_key]
+
+    # Expose clear method of cache to wrapped function
+    cache_aware_func.clear = cache.clear
+
+    return cache_aware_func
+
+
+@map_versionsuffixes_cache
+def map_common_versionsuffixes(software_name, software_version_stub, original_toolchain, toolchain_mapping):
+    """
+    Create a mapping of common versionssuffixes (like `-Python-%(pyvers)`) between toolchains
```

Comment:
If the `versionsuffix` is well written, there should be no need for this actually... If `%(pyver)s` is used, that should just follow the version of the `Python` in `dependencies`. What am I missing here?

How about just failing hard if there's a hardcoded version found in `versionsuffix`?

Reply:
We are (I think) working with a fully parsed which means the template has been resolved (I might be wrong here). In the dump method if the python version doesn't match that in the prefix (which happens if we update the python version) I imagine that the template isn't put back in place...am I wrong?

Reply:
Actually this is indeed required. At the very least I need the mapping to know how to search correctly for possible matches in `search_easyconfigs` there I need to know the actual version of Python in the version suffix, i.e., if I am updating Python to 3.7.0 I can only look for possible matches in the target toolchain with version suffix `-Python-3.7.0`